apps/daq_application.cxx

```diff
@@ -79,19 +79,34 @@ main(int argc, char* argv[])
     exit(0);
   }
 
-  // Set/Update the application and partition name in the environment. Used by logging/ers.
-  setenv("DUNEDAQ_APPLICATION_NAME", args.app_name.c_str(), 0);
-  setenv("DUNEDAQ_PARTITION", args.session_name.c_str(), 0);
+  // Get the application and session name from the environment.
+  std::string app_name = "";
+  std::string session_name = "";
+
+  char* app_name_c = getenv("DUNEDAQ_APPLICATION_NAME");
+  char* session_name_c = getenv("DUNEDAQ_SESSION");
+
+  if (!app_name_c || strcmp(app_name_c, app_name.c_str()) == 0) {
```

Comment:
What is this check for?
```
strcmp(app_name_c, app_name.c_str()) == 0
```
To ensure that the variable is not empty?

Reply:
Yes, sorry, my C++ is a bit rusty